### PR TITLE
feat: Update for populate exchange

### PR DESCRIPTION
**Pull request by imranolas**
Adds the populate exchange and updates mutations to use it

## Prerequisites
- Requires deployment of https://github.com/kitten/threed-example-api/pull/2
- Requires urql exchange release of:
  - https://github.com/FormidableLabs/urql-exchange-graphcache/pull/151
  - https://github.com/FormidableLabs/urql-exchange-graphcache/pull/150

### CODE REVIEW

File: introspect-schema [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/env node
+
+const { introspectionQuery } = require('graphql');
```
Comment (imranolas) [inline introspect-schema:3]: This executable requests the introspected schema and writes it to `src/build` for use in the populate exchange. It is triggered pre build/start in package.json

File: src/index.js [hunk 2]
```diff
@@ -230,9 +205,11 @@ const cache = cacheExchange({
 });
 
 const client = createClient({
-  url: "https://threed-test-api.herokuapp.com/graphql",
+  url: 'http://localhost:3000/graphql',
   exchanges: [
     dedupExchange,
+    // devtoolsExchange,
+    populateExchange({ schema }),
```
Comment (imranolas) [inline src/index.js:212]: Add populate exchange

File: src/index.js [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
-import { GlobalStyles } from "./layout/GlobalStyles";
-import { getToken, setToken } from "./utils/auth";
-import { ME_QUERY } from "./modules/auth/meQuery";
-import { THREAD_FRAGMENT } from "./modules/threads/fragments";
+import App from './App';
+import { GlobalStyles } from './layout/GlobalStyles';
+import { getToken, setToken } from './utils/auth';
+import { ME_QUERY } from './modules/auth/meQuery';
+import { THREAD_FRAGMENT } from './modules/threads/fragments';
+import schema from './build/schema.json';
```
Comment (imranolas) [inline src/index.js:15]: importing the introspected schema

File: src/modules/threads/create/index.js [hunk 4]
```diff
@@ -71,10 +74,9 @@ const StyledButton = styled(Button)`
 const CREATE_THREAD_MUTATION = gql`
   mutation($title: String!, $text: String!) {
     createThread(input: { title: $title, text: $text }) {
-      ...ThreadFragment
+      node @populate
```
Comment (imranolas) [inline src/modules/threads/create/index.js:77]: Populate directive replaces mutation body

File: src/modules/threads/detail/reply/Create.js [hunk 5]
```diff
@@ -47,8 +50,10 @@ const StyledButton = styled(Button)`
 const CREATE_REPLY_MUTATION = gql`
   mutation($threadId: ID!, $text: String!) {
     reply(input: { threadId: $threadId, text: $text }) {
-      id
-      likesNumber
+      node @populate
```
Comment (imranolas) [inline src/modules/threads/detail/reply/Create.js:53]: Populate directive replaces mutation body

File: src/modules/threads/detail/reply/index.js [hunk 6]
```diff
@@ -53,16 +50,18 @@ const Text = styled.p`
 `;
 
 const LIKE_REPLY = gql`
-  mutation ($id: ID!) {
-    likeReply (replyId: $id) {
-      ...ReplyFragment
+  mutation($id: ID!) {
+    likeReply(replyId: $id) {
+      node @populate {
```
Comment (imranolas) [inline src/modules/threads/detail/reply/index.js:55]: Populate directive replaces mutation body

File: src/modules/threads/list/Thread.js [hunk 7]
```diff
@@ -71,11 +71,9 @@ const TextGroup = styled.div`
 const LIKE_THREAD_MUTATION = `
   mutation($id: ID!) {
     likeThread(threadId: $id) {
-      id
-      hasUserLiked
-      likesNumber
+      node @populate
```
Comment (imranolas) [inline src/modules/threads/list/Thread.js:74]: Populate directive replaces mutation body

**Review by imranolas — COMMENTED** [review]
Sorry for the big diff – it's mostly prettier formatting. I've annotated the primary changes to clarify for the reviewer.

File: package.json [hunk 8]
```diff
@@ -16,8 +16,9 @@
     "urql": "1.7.0"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
+    "introspect": "./introspect-schema",
+    "start": "yarn introspect && react-scripts start",
+    "build": "yarn introspect && react-scripts build",
```
Comment (JoviDeCroock) [inline package.json:21]:
```suggestion
    "build": "NODE_ENV=production yarn introspect && react-scripts build",
```

Fixes the issue seen in CI

**Pull request closed without merging**